src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp

```diff
@@ -511,13 +511,14 @@ QMCCostFunctionBatched::Return_rt QMCCostFunctionBatched::correlatedSampling(boo
   bool compute_nlpp             = useNLPPDeriv && (includeNonlocalH != "no");
   bool compute_all_from_scratch = (includeNonlocalH != "no"); //true if we have nlpp
 
+  size_t opt_num_crowds = walkers_per_crowd_.size();
   // Divide samples among crowds
-  std::vector<int> samples_per_crowd(opt_num_crowds_ + 1);
-  FairDivide(rank_local_num_samples_, opt_num_crowds_, samples_per_crowd);
+  std::vector<int> samples_per_crowd(opt_num_crowds + 1);
+  FairDivide(rank_local_num_samples_, opt_num_crowds, samples_per_crowd);
```

Comment:
This may cause larger than expected imbalance when crowd sizes are not equal. Need to make sure per-walker number of samples differ by one at maximum instead of the number of per-crowd samples differ by 1.

Reply:
I added a TODO comment to the same code in checkConfigurations.  Could this change be moved to a future PR?

Reply:
Yes. But please open an issue to track this.

Reply:
Could you open an github issue? I think this the last thing blocking this PR (from my perspective).

Reply:
Opened #3888